core/sql/optimizer/ScanOptimizer.cpp

```diff
@@ -374,8 +399,21 @@ static NABoolean checkMDAMadditionalRestriction(const ColumnOrderList& keyPredsB
 
      }
    }     
-   if(presentKeyColumns > noOfmissingKeyColumns) return TRUE;
-   return FALSE;
+   switch ( strategy ) {
+     case MAJORITY_WITH_PREDICATES:
+        return (presentKeyColumns > noOfmissingKeyColumns);
```

Comment:
I am not sure I see the point of this strategy. Suppose I have a 6 column key and a query with predicates on 4 columns. It makes a huge difference whether the predicates are on the leading columns (better) vs. the trailing columns (worse). This heuristic does not take the column order into account.